client/controllers/searchInput.coffee

```diff
@@ -44,6 +45,15 @@ Template.searchInput.events
       instance.textFilter.set
         $regex: instance.$(event.target).val()
         $options: 'i'
+      if instance.data.tableId
+        Meteor.setTimeout(->
+          count = parseInt($("##{instance.data.tableId}").next().find('span.rows-per-page-count').text(), 10)
+          if count == 0
+            # hide loading indicator
+            $('.loading').hide()
+          else
+            $('.loading').show()
+        , 0)
```

Comment:
The loading indicator would be partially shown if there was one row in the table, so I changed it to be <= 1.  AFAIK meteor.defer is the same thing as `Meteor.setTimeout ... , 0` and the syntax is cleaner especially with coffeescript.